poet2-validator-integration.yaml

```diff
@@ -0,0 +1,164 @@
+      dockerfile: sawtooth-core/sdk/examples/intkey_python/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./sawtooth-core:/project/sawtooth-core
+    working_dir: '/project/sawtooth-core'
+    container_name: intkey-tp-python-local
+    depends_on:
+      - validator
+    command: |
+      bash -c "
+        bin/protogen
+        cd sdk/examples/intkey_python
+        python3 setup.py clean --all
+        python3 setup.py build
+        intkey-tp-python -vv -C tcp://validator:4004
+      "
+    stop_signal: SIGKILL
+
+  client:
+    build:
+      context: .
+      dockerfile: sawtooth-core/docker/sawtooth-shell
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./sawtooth-core:/project/sawtooth-core
+    working_dir: '/project/sawtooth-core'
+    container_name: sawtooth-shell-local
+    depends_on:
+      - validator
+    command: |
+      bash -c "
+        sawtooth keygen &&
+        tail -f /dev/null
+      "
+    stop_signal: SIGKILL
+
+  validator:
+    build:
+      context: .
+      dockerfile: sawtooth-core/validator/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./sawtooth-core:/project/sawtooth-core
+    working_dir: '/project/sawtooth-core'
+    container_name: sawtooth-validator-local
+    expose:
+      - 4004
+      - 8800
+    ports:
+      - "4004:4004"
+    # start the validator with an empty genesis batch
+    command: |
+      bash -c "
+        sawadm keygen --force && \
+        sawadm genesis && \
+        sawtooth-validator -vv \
+            --endpoint tcp://validator:8800 \
+            --bind component:tcp://eth0:4004 \
+            --bind network:tcp://eth0:8800 \
+            --bind consensus:tcp://eth0:5005 \
+      "
+    stop_signal: SIGKILL
+
+
+  rest-api:
+    build:
+      context: .
+      dockerfile: sawtooth-core/rest_api/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./sawtooth-core:/project/sawtooth-core
+    working_dir: '/project/sawtooth-core'
+    container_name: sawtooth-rest-api-local
+    ports:
+      - "8008:8008"
+    depends_on:
+      - validator
+    command: |
+      bash -c "
+        bin/protogen
+        cd rest_api
+        python3 setup.py clean --all
+        python3 setup.py build
+        sawtooth-rest-api -v --connect tcp://validator:4004 --bind rest-api:8008
+      "
+    stop_signal: SIGKILL
+
+  poet-2-engine:
+    build:
+      context: .
+      dockerfile: poet2-engine.dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./:/project/sawtooth-poet2
+    working_dir: '/project/sawtooth-poet2/src/core'
+    container_name: poet-2-engine
+    depends_on:
+      - validator
+    entrypoint: "bash -c \"\
+        cargo run --bin poet2 -- -C tcp://validator:5005 --verbose \""
```

Comment:
Update port to 5050.

Reply:
Updated the port 5050 in poet2-validator-integration.yaml which would be default one.